vault/logical_system_activation_flags_test.go

```diff
@@ -0,0 +1,492 @@
+func TestSystemBackend_activationFlagsPaths(t *testing.T) {
+	printPaths := func(paths []*framework.Path) string {
+		list := ""
+		for _, p := range paths {
+			list += fmt.Sprintf("Pattern: %s\n", (*p).Pattern)
+		}
+		return list
+	}
+
+	core, _, _ := TestCoreUnsealedWithConfig(t, &CoreConfig{})
+
+	tests := map[string]struct {
+		settingAdminNSPath  bool
+		enableHVDTierEnvVar bool
+		enableOnHVDEnvVar   bool
+		wantedPathCount     int
+	}{
+		"if registering paths to the root namespace then expect 2 paths": {
+			settingAdminNSPath:  false,
+			enableHVDTierEnvVar: false,
+			enableOnHVDEnvVar:   false,
+			wantedPathCount:     2,
+		},
+		"if registering paths to the root namespace with EnvHVDTierSecretsSync set then expect 2 paths": {
+			settingAdminNSPath:  false,
+			enableHVDTierEnvVar: true,
+			enableOnHVDEnvVar:   false,
+			wantedPathCount:     2,
+		},
+		"if registering paths to the root namespace on HVD then expect 2 paths": {
+			settingAdminNSPath: false,
+			enableOnHVDEnvVar:  true,
+			wantedPathCount:    3,
+		},
+		"if registering paths to the admin namespace without EnvHVDTierSecretsSync set then expect 1 path": {
+			settingAdminNSPath:  true,
+			enableHVDTierEnvVar: false,
+			enableOnHVDEnvVar:   false,
+			wantedPathCount:     1,
+		},
+		"if registering paths to the admin namespace with EnvHVDTierSecretsSync set then expect 1 path": {
+			settingAdminNSPath:  true,
+			enableHVDTierEnvVar: true,
+			enableOnHVDEnvVar:   false,
+			wantedPathCount:     1,
+		},
+		"if registering paths to the admin namespace with EnvHVDFlag set then expect 1 path": {
+			settingAdminNSPath: true,
+			enableOnHVDEnvVar:  false,
+			wantedPathCount:    1,
+		},
+		"if registering paths to the admin namespace with EnvHVDTierSecretsSync and on HVD then expect 2 paths": {
+			settingAdminNSPath:  true,
+			enableHVDTierEnvVar: true,
+			enableOnHVDEnvVar:   true,
+			wantedPathCount:     3,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			if tt.enableHVDTierEnvVar {
+				t.Setenv(EnvHVDTierSecretsSync, "enable")
+			}
+			if tt.enableOnHVDEnvVar {
+				t.Setenv(activationflags.EnvHVDFlag, "set")
+			}
+
+			got := core.systemBackend.activationFlagsPaths(tt.settingAdminNSPath)
+			gotLength := len(got)
+			if gotLength != tt.wantedPathCount {
+				t.Errorf("Received %d paths back from SystemBackend.activationFlagsPaths(), wanted %d paths.\nReceived the following path patterns back:\n%s", gotLength, tt.wantedPathCount, printPaths(got))
+			}
+		})
+	}
+}
+
+// TestActivationFlags_Write tests the write operations for the activation flags, for both
+// activate and deactivate, whether we're running in Enterprise or HVD, and whether we're on
+// a plus tier or not when in HVD.
+func TestActivationFlags_Write(t *testing.T) {
+	testActivationFlags_Write_Activate(t)
+	testActivationFlags_Write_Deactivate(t)
+}
+
+// TestActivationFlags_Invalidate: Tests State Change Logic for Flags
```

Comment:
Should this Invalidate test be in CE? It looks like the Invalidate methods have been kept in Enterprise so I'd guess this test will fail in CE right?